moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -461,6 +470,32 @@ public final void setFailOnUnknown(boolean failOnUnknown) {
     }
   }
 
+  /**
+   * Returns a new {@code JsonReader} that can read data from this {@code JsonReader} without
```

Comment:
nice! will we need a note to say that behavior is undefined if the original gets ahead of the peeked reader?

Reply:
yeah, mirroring with Okio is perfect!